src/main/java/datawave/query/util/TypeMetadata.java

```diff
@@ -103,6 +104,54 @@ public Collection<String> getTypeMetadata(String fieldName, String ingestType) {
         return Sets.newHashSet(map.get(fieldName));
     }
     
+    /**
+     * Returns a set of all Normalizer names associated with the given fieldName
+     *
+     * @param fieldName
+     * @return
+     */
+    public Set<String> getNormalizerNamesForField(String fieldName) {
+        if (fieldName == null || fieldName.isEmpty()) {
+            return Collections.emptySet();
+        }
+        
+        Set<String> normalizers = new HashSet<>();
+        for (Multimap<String,String> entry : this.typeMetadata.values()) {
+            normalizers.add(String.valueOf(entry.get(fieldName)));
+        }
+        
+        return normalizers;
+    }
+    
+    public Set<String> getNormalizerNamesForField() {
+        return getNormalizerNamesForField(null);
+    }
```

Comment:
What's the purpose of this method?

Reply:
Originally wanted to cover the case of calling the method without any field names but I suppose that doesn't really make sense - will delete and force calling with some kind of field name.